apps/dashboard/app/(app)/authorization/permissions/page.tsx

```diff
@@ -55,6 +63,23 @@ export default async function RolesPage() {
     permission.keys = permission.keys.filter(({ key }) => key.deletedAt === null);
     return permission;
   });
+
+  type Permission = (typeof workspace.permissions)[number];
+  const getKeys = (p: Permission) => {
```

Comment:
we should not call this twice.
it doesn't even have to be a function I think. 
just do it all in the parent scope and then use it as `connectedKeys.size` in the jsx below.

Reply:
The problem is we are passing the permissions down to the cards via .map, I actually thought of making the cards a separate component. But don't know if that's worth it.